src/foam/core/JSON.js

```diff
@@ -459,6 +497,57 @@ foam.CLASS({
 });
 
 
+foam.CLASS({
+  package: 'foam.json',
+  name: 'JSONParserStringifier',
+  extends: 'foam.json.Outputter',
+  implements: [
+    'foam.json.Parser',
+    'foam.json.Stringifier'
+  ],
+
+  documentation: `An outputter for "proper" JSON (with quoted keys, etc.). Not
+      to be confused with foam.json's standard outputters, which operate over
+      FON.`,
+
+  properties: [
+    {
+      documentation: `Map of {<class id>: true} for allowed classes`,
+      name: 'allowClassMap',
+      value: null
+    },
+    {
+      name: 'creationContext',
+      expression: function(allowClassMap) {
+        var baseCtx = this.__subContext__;
```

Comment:
New implementation is passed a `creationContext`, usually provided by a `foam.box.Context`(which will use `ClassWhitelistContext` if it has a whitelist).